File: lista1/Q3.jl
using DataFrames
using Plots


# Dados Iniciais
# Como termo geral da sequencia, o valor de M, o valor de K
# E um valor inicial N ⋜ Nmax
a(n) = (-1)^(n)
M = 2
K = -2
N = 1
Nmax = 10

an_s = []
n_s = []
for i = 1:Nmax
   append!(an_s, a(i))
   append!(n_s, i)
end

df = DataFrame(n = n_s, an = an_s)
print(df)
x = N: 1 : Nmax
y_M = [M for i = x]
y_K = [K for i = x]
y_an = [a(i) for i = x]
Y = [y_M, y_K, y_an]

plot(x, Y, label=["M" "N" "an"], lw=5, ls=[:solid :solid :dot])

File: lista4/Q3.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt



# Termo geral da serie
def a(n):
    return 1/(np.sqrt(n+1)*np.sqrt(n)*(np.sqrt(n+1)+np.sqrt(n)))


# Dados iniciais
Kmin = 1
Kmax = 15

k = []
Sk = []
Sk_total = 0

for n in range(Kmin, Kmax+1):
    Sk_total += a(n) 
    k.append(n)
    Sk.append(Sk_total)


# Abaixo temos como printar os dados em uma tabela
# # Por padrão vou deixar comentando essa parte
# columns = ["a(n)"] # Nomenclatura da coluna com os termos de a(n)

# # Abaixo estamos criando um objeto do tipo dataframe, onde é passado os dados sendo Sk
# # e o indices senod k
# df = pd.DataFrame(data = Sk, columns=columns, index=k) 

# print(df)

# Plot do Grafico da sequencia
legend = ['Sk']

plt.scatter(x=k, y=Sk)
plt.legend(legend)
plt.show()


